src/test/java/reposense/authorship/AnnotatorAnalyzerTest.java

```diff
@@ -0,0 +1,79 @@
+package reposense.authorship;
+
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import reposense.authorship.model.FileInfo;
+import reposense.authorship.model.FileResult;
+import reposense.authorship.model.LineInfo;
+import reposense.model.Author;
+import reposense.parser.InvalidLocationException;
+import reposense.template.GitTestTemplate;
+import reposense.util.TestUtil;
+
+public class AnnotatorAnalyzerTest extends GitTestTemplate {
+
+    private static final Author MAIN_AUTHOR = new Author(MAIN_AUTHOR_NAME);
+    private static final Author FAKE_AUTHOR = new Author(FAKE_AUTHOR_NAME);
+    private static final Date SINCE_DATE = TestUtil.getSinceDate(2018, Calendar.FEBRUARY, 8);
+    private static final Date UNTIL_DATE = TestUtil.getUntilDate(2018, Calendar.FEBRUARY, 10);
+
+    public FileResult getFileResult(String relativePath) {
+        FileInfo fileInfo = FileInfoExtractor.generateFileInfo(config.getRepoRoot(), relativePath);
+        return FileInfoAnalyzer.analyzeFile(config, fileInfo);
+    }
+
+    /**
+     * Asserts the correctness of authorship overriding with regards to @@author tags given
+     * in the test file.
+     */
+    public void assertAnnotationAnalysisCorrectness(FileResult fileResult, Author originalAuthor,
+            Author overridingAuthor) {
+        for (LineInfo line : fileResult.getLines()) {
+            Author lineAuthor = line.getAuthor();
+            if (line.getContent().startsWith("fake")) {
+                Assert.assertEquals(originalAuthor, lineAuthor);
+            } else {
+                Assert.assertEquals(overridingAuthor, lineAuthor);
+            }
+        }
+    }
+
+    @Before
+    public void before() throws InvalidLocationException {
+        super.before();
```

Comment:
I have overriden the `before` method in `GitTestTemplate` because I have some additional logic to perform before each test case. Specifically, I need to set the same sinceDate and untilDate for the config, before each test case.

Reply:
Looks like you need to add the `org.junit.jupiter.api` dependency for it. Don't know if its a feasible option. Can ask the senior devs.

Reply:
Looks fine here.

Reply:
> JUnit provides a BeforeEach() method. So if it needs to run before each method you can use that, instead of over-riding?

In this case, the logic only needs to be executed once before all tests.